src/core/frames/frame_controller.ts

```diff
@@ -432,6 +426,20 @@ export class FrameController
     return !event.defaultPrevented
   }
 
+  private handleFrameMissingFromResponse(fetchResponse: FetchResponse) {
+    this.showFrameMissingError()
+    this.throwFrameMissingException(fetchResponse)
+  }
+
+  private showFrameMissingError() {
```

Comment:
Consider renaming to `renderFrameMissingError` to stick with the "render" language, and to suggest that if we're going to do any heavier rendering, we should delegate to the frame's view, [using an ErrorRenderer](https://github.com/hotwired/turbo/blob/b3facd8d486f89f0104baa7c64390f4990c72d79/src/core/drive/page_view.ts#L32-L36) as we do for [page visits](https://github.com/hotwired/turbo/blob/b3facd8d486f89f0104baa7c64390f4990c72d79/src/core/drive/navigator.ts#L113).

I could be convinced we should use a renderer now, it just seemed a bit too heavy. Thoughts?

Reply:
Agreed on the rename 👍

I'd be inclined not to use a renderer here, as long as we don't have anything more involved to do that set some text. But I also could be convinced otherwise :)

Reply:
Or, maybe without a renderer it should at least use a view method to set the content? Might be a lightweight middle ground. I'll take a look at this tomorrow.

Reply:
Good call. It would fit well in `FrameView`, which also sets `innerHTML` to invalidate the frame. Could add a `missing()` to go with `invalidate()`.